emcc.py

```diff
@@ -2956,9 +2956,20 @@ def phase_binaryen(target, options, wasm_target):
     logger.warning("Wasm source map won't be usable in a browser without --source-map-base")
   # whether we need to emit -g (function name debug info) in the final wasm
   debug_info = settings.DEBUG_LEVEL >= 2 or options.profiling_funcs
-  # whether we need to emit -g in the intermediate binaryen invocations (but not necessarily at the very end).
-  # this is necessary for emitting a symbol map at the end.
-  intermediate_debug_info = bool(debug_info or options.emit_symbol_map or settings.ASYNCIFY_ONLY or settings.ASYNCIFY_REMOVE or settings.ASYNCIFY_ADD)
+  # whether we need to emit -g in the intermediate binaryen invocations (but not
+  # necessarily at the very end). this is necessary if we depend on debug info
+  # during compilation, even if we do not emit it at the end.
+  # we track the number of causes for needing intermdiate debug info so
+  # that we can stop emitting it when possible - in particular, that is
+  # important so that we stop emitting it before the end, and it is not in the
+  # final binary (if it shouldn't be)
+  intermediate_debug_info = 0
+  if debug_info:
+    intermediate_debug_info += 1
+  if options.emit_symbol_map:
```

Comment:
Include `and os.path.exists(wasm_target)` here .. then each +1 there is linked to binaryen run that always runs ?

Reply:
We can't do that here: the wasm target exists at this point, but is removed later on. That is, we remove the wasm target when we stop needing it, which is somewhere in the middle of the process.